lib/_http_client.js

```diff
@@ -172,6 +172,13 @@ function ClientRequest(input, options, cb) {
     method = this.method = 'GET';
   }
 
+  const maxHeaderSize = options.maxHeaderSize;
+  if (maxHeaderSize !== undefined && (!Number.isSafeInteger(maxHeaderSize) ||
```

Comment:
Nit: would `validateInteger` from validators.js fit here? If not, I think we should use `NumberIsSafeInteger` from `primordials` instead of global Number, though I'm not sure.

Reply:
Sure, done.